Game_Development/Dodge-the-Blocks/main.py

```diff
@@ -0,0 +1,314 @@
+            player.rect.y += 3
+
+def display_score(screen, score, wall_touch_count, level):
+    font = pygame.font.SysFont('comicsansms', 30)
+    score_text = font.render(f"Score: {score - wall_touch_count}", True, DARK_BLUE)
+    level_text = font.render(f"Level: {level}", True, DARK_BLUE)
+    screen.blit(score_text, [10, 10])
+    screen.blit(level_text, [10, 50])
+
+def move_blocks(block, index, l1, l2):
+    block.rect.x += l1[index]
+    block.rect.y += l2[index]
+    if block.rect.x >= 840 or block.rect.x < 10:
+        l1[index] *= -1
+    if block.rect.y >= 540 or block.rect.y < 10:
+        l2[index] *= -1
+
+
+# Block class definition
+class Block(pygame.sprite.Sprite):
+    
+    def __init__(self, color, width, height):
+        
+        super().__init__()
+        
+        self.image = pygame.Surface([width, height])
+        self.image.fill(color)
+ 
+        self.rect = self.image.get_rect()
+
+
+# Main game loop and logic
+pygame.init()
+pygame.mixer.init()
+size = (900,600)
+screen = pygame.display.set_mode((size))
+pygame.display.set_caption("My Game")
+
+pygame.mixer.music.load(welcome_screen_music)
+start_screen(screen)
+pygame.mixer.music.stop()
+pygame.mixer.music.load(in_game_music)
+pygame.mixer.music.play(-1)
+
+# Main game loop and logic
+pygame.init()
+size = (900,600)
+screen = pygame.display.set_mode((size))
+pygame.display.set_caption("My Game")
+
+
+# Game variables
+block_list = pygame.sprite.Group()                       #for the obstacle blocks
+all_sprites_list = pygame.sprite.Group()                 #for the obstacle blocks and the player block
+l1 = []
+l2 = []
+
+
+# Setup obstacle blocks
+for i in range(1):
+    block = Block(RED, 50, 50)
+    block.rect.x = random.randrange(200,830)
+    block.rect.y = random.randrange(200,530)
+    block_list.add(block)
+    all_sprites_list.add(block)
+
+for k in range(15):
+    xchange = random.randrange(0, 5)
+    l1.append(xchange)
+    ychange = random.randrange(0, 5)
+    l2.append(ychange)
+
+# Setup player
+player = Block(GREEN, 20, 20)
+player.rect.left = 11
+player.rect.top = 11
+all_sprites_list.add(player)
+
+# Game loop
+game_loop = True
+clock = pygame.time.Clock()
+wall_touch_count = 0 
+score = 100
+level = 1
+
+while game_loop:
+    for event in pygame.event.get():
+        if event.type == pygame.QUIT:
+            game_loop = False
+
+
+    # Create the game world (bg and walls)
+    screen.fill(LIGHT_BLUE)
+    pygame.draw.line(screen,BLACK,[0,0],[0,600],20)
+    pygame.draw.line(screen,BLACK,[0,0],[900,0],20)
+    pygame.draw.line(screen,BLACK,[0,600],[900,600],20)
+    pygame.draw.line(screen,BLACK,[900,0],[900,500],20)
+    display_score(screen, score, wall_touch_count, level)
+
+
+    # Collision detection
+    blocks_hit_list = pygame.sprite.spritecollide(player, block_list, True)
+    all_sprites_list.draw(screen)
+    pygame.display.flip()
+
+
+    # Player movement
+    handle_player_movement(event, player)

```

Comment:
The function ```handle_player_movement(event, player)``` is called outside the event loop. Due to this only last event is repeating (i.e. Start of the game) and it crashes afterwards. Please update this and will merge once updated.

Reply:
That won't work. When the `handle_player_movement(event, player)` function is called inside the event loop, it only processes discrete events. Pygame doesn’t continuously repeat key events while a key is held down, but rather treats it as individual key presses that need to be handled explicitly.

So, what would happen if the function is called inside the event loop?
Continuing to press on an arrow key only moves it by one unit and then it stops. For example if i hold down the _down arrow key_ then the player block moves one unit down and then stops. If i want to move further down, i have to lift up my finger and then press the down key again. This is annoying and not the desired functionality.

Why keep the function outside the event loop?
It keeps checking for the event in every frame, meaning it continually processes movement as long as the key is pressed, which is what you want for smooth movement. This is the desired functionality.